### ISSUE TITLE
NVDA fails to report some formatted text when navigating by word or character in Thunderbird

### ISSUE THREAD

**Issue by SaschaCowley** [nvaccess/nvda#16840]
### Steps to reproduce:

1. Create a new HTML message in Thunderbird.
2. Insert the following text, applying bold to the word "bold" (italics and underline also exhibit this behaviour)
   > This is bold
3. Attempt navigating by character (left and right arrows) or word (control and left and right arrows).

#### Navigation by character

```
cap   T
right arrow
h
right arrow
i
right arrow
s
right arrow
space
right arrow
i
right arrow
s
right arrow
space
right arrow
blank
right arrow
o
right arrow
l
right arrow
d
right arrow
blank
right arrow
blank
left arrow
dot
left arrow
d
left arrow
l
left arrow
o
left arrow
b
left arrow
space
left arrow
s
left arrow
i
left arrow
space
left arrow
s
left arrow
i
left arrow
h
left arrow
cap   T
```

#### Navigation by word

```
This
ctrl+right arrow
is 
ctrl+right arrow
bold.
ctrl+right arrow
bold.
ctrl+left arrow
is 
ctrl+left arrow
is 
ctrl+left arrow
This 
```

### Expected behavior:

That the navigation commands work identically, whether or not the text has been formatted.

#### NVDA installed/portable/running from source:

Installed

#### NVDA version:

alpha-32834,6d8323c9 (2024.4.0.32834)

#### Windows version:

23H2 (OS Build 22635.3790)

#### Name and version of other software in use when reproducing the issue:

Thunderbird 115.12.2 (64-bit)

#### Does the issue still occur after restarting your computer?

Yes

#### Have you tried any other versions of NVDA? If so, please report their behaviors.

No

#### If NVDA add-ons are disabled, is your problem still occurring?

Yes

#### Does the issue still occur after you run the COM Registration Fixing Tool in NVDA's tools menu?

Yes

**Comment by Adriani90** [comment]
So it seems like due to the formating of the text when pressing right arrow, NVDA interpres some things like being a linefeed character? Or a blank line. Is my understanding correct?

cc: @emitche, @thgcode maybe you can see whether this is reproducible on your side as well?
